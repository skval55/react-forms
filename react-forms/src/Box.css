.Box{
    display: flex;
    align-items: end;
}


.Box-box{
    border-radius: 5%;
    border: 1px solid black;
}



.Box-button {
    height: 30px;
    width: 30px;
}