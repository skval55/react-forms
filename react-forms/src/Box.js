import React from "react";
import "./Box.css";

const Box = ({ code, handleDelete, height, width, backgroundColor }) => {
  const boxStyle = {
    backgroundColor: backgroundColor,
    height: height,
    width: width,
  };
  return (
    <div className="Box">
      <div index={code} className="Box-box" style={boxStyle}></div>
      <button className="Box-button" onClick={() => handleDelete(code)}>
        X
      </button>
    </div>
  );
};

export default Box;
