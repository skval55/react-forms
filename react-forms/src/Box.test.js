import { render } from "@testing-library/react";
import Box from "./Box";

// smoke test
test("renders without crashing", () => {
  render(
    <Box
      code="test code"
      height={60}
      width={60}
      backgroundColor="black"
      handleDelete={() => console.log("test")}
    />
  );
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(
    <Box
      code="test code"
      height={60}
      width={60}
      backgroundColor="black"
      handleDelete={() => console.log("test")}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
