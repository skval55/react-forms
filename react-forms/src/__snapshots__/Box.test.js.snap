// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      index="test code"
      style="background-color: black; height: 60px; width: 60px;"
    />
    <button
      class="Box-button"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
